### DISCUSSION TITLE
ZLUDA with DAZ Studio and/or NVidia iRay?

### DISCUSSION THREAD

**Discussion by mmn109** [vosen/ZLUDA#83] [Q&A]
Has anyone tried ZLUDA with DAZ Studio and/or NVidia iRay?  DAZ Studio is a free product that uses iRay as its render engine.  It has a good sized user-base.  Check it out at daz3d.com.

**Comment by nico-haeusser** [comment]
I have tested it a bit.
Zluda works amazingly well in Blender with Cuda. 
In DAZ the graphics card is recognized, but when rendering iray gets an error due to missing Nvidia drivers and cannot access the graphics card.

**Comment by Nughu** [comment]
I've tried running DAZ Studio using `zluda -- DAZStudio.exe`, but it makes DAZ crash immediately with the following error:
`DAZStudio.exe caused UNKNOWN_ERROR in module "C:\Windows\System32\KERNELBASE.dll" at 0033:0000000017FC4ED9, RaiseException()+105 byte(s)`

**Reply by Nughu** [reply]
I've now gotten it to run by using `set HIP_VISIBLE_DEVICES=1` before zluda itself. Sadly though, Iray still just uses my CPU.

**Reply by nico-haeusser** [reply]
You can turn off "Allow CPU Fallback". This way it only uses the GPU.

But for me then ist gives something like this in the log files:
`2024-02-17 16:26:16.000 [WARNING] :: \src\pluginsource\DzIrayRender\dzneuraymgr.cpp(376): Iray [ERROR] - IRAY:RENDER ::   1.0   IRAY   rend error: optixInit() failed: Library not found. Please update your NVIDIA driver (www.nvidia.com) to at least 522.06.
2024-02-17 16:26:16.000 [WARNING] :: \src\pluginsource\DzIrayRender\dzneuraymgr.cpp(376): Iray [WARNING] - IRAY:RENDER ::   1.0   IRAY   rend warn : CUDA device 0 (AMD Radeon RX 6700 XT [ZLUDA]) is no longer available for rendering.`

**Comment by mmn109** [comment]
Thanks for sharing your experience with Zluda and DAZ Studio, nico-haeusser and Nughu.  

Too bad the error message does not tell you which library cannot be found.

**Reply by nico-haeusser** [reply]
Right now I need my system to be stable for 1-2 weeks. If no one else has a solution then I can mess around with drivers and maybe find a solution.

**Comment by Nughu** [comment]
In my case, DAZ doesn't even show any GPU under IRay Render settings. updated daz but still the same

**Comment by mmn109** [comment]
In the DAZ forums, people are saying that Vluda will not run on Daz Studio/iRay because Vluda has no support for Optix.  Here is Daz's explanation of Optix: https://www.daz3d.com/forums/discussion/319286/iray-vs-optix

Here is a discussion of Vluda on the Daz forums: https://www.daz3d.com/forums/discussion/673711/cuda-can-now-run-on-amd-gpus/p1

**Reply by nico-haeusser** [reply]
The discussion in the daz forum is how I found out about Zluda :D

There is minimum support for optix in Zluda in Linux?
Maybe I will play around with Zluda in Linux.

**Comment by toxieainc** [comment]
@vosen Is there the possibility to configure the compute/SM version of the 'emulated' CUDA GPU? Cause here it could work to switch to SM 7.0 or earlier as then no OptiX/RT cores are used.

**Comment by kanav-khanna** [comment]
I believe we might be able to get Daz 3D working via ZLUDA using Octane Render.
By default, Daz 3D’s built-in renderer continues to use the CPU even though it detects my AMD GPU.
Octane Render is a plugin for Daz that allows GPU rendering, which may serve as a workaround to get Daz to use AMD more effectively.
To my knowledge, Octane Render uses its own denoising implementation rather than NVIDIA’s OptiX. Since more recent versions of ZLUDA no longer support OptiX, this could work to our advantage.
